src/serde.rs

```diff
@@ -0,0 +1,269 @@
+        state.serialize_field("kernel_version", &self.kernel_version())?; // Option<String>
+        state.serialize_field("os_version", &self.os_version())?; // Option<String>
+        state.serialize_field("long_os_version", &self.long_os_version())?; // Option<String>
+        state.serialize_field("distribution_id", &self.distribution_id())?; // String
+        state.serialize_field("host_name", &self.host_name())?; // Option<String>
+
+        state.end()
+    }
+}
+*/
+
+impl Serialize for dyn NetworkExt {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("Network", 1)?;
+
+        state.serialize_field("received", &self.received())?;
+        state.serialize_field("total_received", &self.total_received())?;
+        state.serialize_field("transmitted", &self.transmitted())?;
+        state.serialize_field("total_transmitted", &self.total_transmitted())?;
+        state.serialize_field("packets_received", &self.packets_received())?;
+        state.serialize_field("total_packets_received", &self.total_packets_received())?;
+        state.serialize_field("packets_transmitted", &self.packets_transmitted())?;
+        state.serialize_field(
+            "total_packets_transmitted",
+            &self.total_packets_transmitted(),
+        )?;
+        state.serialize_field("errors_on_received", &self.errors_on_received())?;
+        state.serialize_field("total_errors_on_received", &self.total_errors_on_received())?;
+        state.serialize_field("errors_on_transmitted", &self.errors_on_transmitted())?;
+        state.serialize_field(
+            "total_errors_on_transmitted",
+            &self.total_errors_on_transmitted(),
+        )?;
+        state.serialize_field("mac_address", &self.mac_address())?;
+
+        state.end()
+    }
+}
+
+impl Serialize for dyn ComponentExt {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("Component", 4)?;
+
+        state.serialize_field("temperature", &self.temperature())?;
+        state.serialize_field("max", &self.max())?;
+        state.serialize_field("critical", &self.critical())?;
+        state.serialize_field("label", &self.label())?;
+
+        state.end()
+    }
+}
+
+impl Serialize for dyn UserExt {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("User", 1)?;
+
+        let uid = *self.id().deref();
+        state.serialize_field("id", &uid)?;
+
+        let gid = *self.group_id().deref();
+        state.serialize_field("group_id", &gid)?;
+
+        state.serialize_field("name", &self.name())?;
+        state.serialize_field("groups", &self.groups())?;
+
+        state.end()
+    }
+}
+
+use crate::{DiskType, DiskUsage, MacAddr, ProcessStatus};
```

Comment:
Please put the imports at the top.

Reply:
Yeah, it's only to see the implementation on structs and the traits on top of the file at the moment

Reply:
Add comments then. I love code comments. :)